datafusion/core/Cargo.toml

```diff
@@ -69,6 +69,13 @@ pyarrow = ["datafusion-common/pyarrow", "parquet"]
 regex_expressions = [
     "datafusion-functions/regex_expressions",
 ]
+recusive-protection = [
```

Comment:
This adds the `recursive-protection` feature to the `datafusion` (core) crate which then activates the feature in the subcrates